### ISSUE TITLE
Failure to register DLL for 32-bit wine

### ISSUE THREAD

**Issue by keantoken** [jhernberg/wineasio#4]
In order to compile wineasio I had to change the shell called by the prepare_64bit_wineasio script. Make created wineasio.dll.so. I'm using Debian Sid.

So I did 

sudo cp wineasio.dll.so /usr/lib/i386-linux-gnu/wine/wineasio.dll.so
sudo cp wineasio.dll.so /usr/lib/x86_64-linux-gnu/wine/wineasio.dll.so

wine64 regsvr32 wineasio.dll
This command succeeds

regsvr32 wineasio.dll
This command fails

I'm not sure, maybe I'm not even supposed to do the second one?

I am trying to install it so I can use this program:

http://www.data-odyssey.nl/Diana.html

It may be buggy and unable to detect wineasio, so I installed Reaper to see if wineasio was working. The wineasio driver is selectable but there are no inputs or outputs. Does this mean wineasio is working and that I need to set up Jack? Or does Reaper just always have that option?

Is wineasio still incompatible with jackd2?

This is like sensory deprivation, maybe it's working and I just can't tell.

**Comment by keantoken** [comment]
Reaper shows selectable inputs and outputs now, although it segfaults in usp10.dll and so is useless.

Diana doesn't detect any inputs/outputs and when it tries to open the control panel it says "erroneous driver number".

**Comment by keantoken** [comment]
64-bit Reaper detects wineasio but 32-bit Reaper does not. So it does look like I need to get 32-bit working somehow. Diana is also 32-bit.

I don't know why it fails to register the 32-bit DLL. I tried compiling with just make, but for some reason it ends up trying to link with 64-bit libraries, using the wrong directories. I don't know how to specify the right ones.

**Comment by michaelsjackson** [comment]
Found any solution? I have same problem in ubuntu 16.04 lts.

**Comment by keantoken** [comment]
It's hard to remember, but I very faintly recall it might be due to the 32 vs 64 bit folders being different on debian vs what the script expects. Find your 32 bit and 64 bit folders and compare that to where wineasio is trying to put the libraries.

**Comment by michaelsjackson** [comment]
wine --version
wine-5.3 (Staging)

whereis wine
wine: /usr/bin/wine /usr/lib/wine /usr/local/lib/wine /opt/wine-stable/bin/wine /opt/wine-devel/bin/wine /opt/wine-staging/bin/wine /usr/share/man/man1/wine.1

regsvr32 wineasio.dll
err:service:process_send_command service protocol error - failed to write pipe!
fixme:service:scmdatabase_autostart_services Auto-start service L"clr_optimization_v4.0.30319_32" failed to start: 1053
regsvr32: DLL 'wineasio.dll' konnte nicht geladen werden


regsvr32 /usr/lib32/wineasio.dll.so
000b:fixme:winediag:start_process Wine Staging 5.3 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
regsvr32: DLL '/usr/lib32/wineasio.dll.so' erfolgreich registriert

Telling successfully registered, but after starting Catia, F5, starting Reaper, Wineasio, still no inputs and outputs are seen, all empty. Hmm.

Any more tips?

**Comment by keantoken** [comment]
At this point I I ran into Issue #5:

https://github.com/jhernberg/wineasio/issues/5

Wineasio connects Jack to WINE programs, do you have Jack set up?

**Comment by michaelsjackson** [comment]
Yes, Jack here. All wineasio problems started after updating ubuntu from 14.04 lts to 16.04 lts, before all was working fine, meaning I could play instruments with little latency, now more latency, less fun. :)  Until this is solved.

**Comment by osirisgothra** [comment]
hmm well the libraries are in the right place yet they dont register anymore for some reason, 64 or 32 bit, the script finds them, but they still fail to register not matter what anymore, i have a feeling there are further restrictions like wine versions that must be considered as it worked before I installed the repositories version.
